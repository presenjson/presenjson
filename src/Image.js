import React from 'react';
import PropTypes from 'prop-types';

const Image = ({ style, src, className, ...rest }) => {
    const styles = {
        backgroundImage: `url('${src}')`,
        ...style
    };

    return <div className={`img ${className}`} style={styles} {...rest} />;
};

Image.propTypes = {
    style: PropTypes.object,
    className: PropTypes.string,
    src: PropTypes.string.isRequired
};

Image.defaultProps = {
    style: {},
    className: ''
};

export default Image;
