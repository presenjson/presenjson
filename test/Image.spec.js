import React from 'react';
import { shallow } from 'enzyme';
import Image from '../src/Image';

describe('Image', () => {
    it('renders correctly', () => {
        const image = shallow(
            <Image
                src="/some/src"
                style={{ margin: 10 }}
                some="prop"
                className="test-class"
            />
        );
        expect(image).toMatchSnapshot();
    });
});
