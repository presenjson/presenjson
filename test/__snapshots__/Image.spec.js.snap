// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image renders correctly 1`] = `
<div
  className="img test-class"
  some="prop"
  style={
    Object {
      "backgroundImage": "url('/some/src')",
      "margin": 10,
    }
  }
/>
`;
